stacks-signer/src/client/stacks_client.rs

```diff
@@ -196,6 +211,138 @@ impl StacksClient {
         Ok(peer_info_data)
     }
 
+    /// Get signer slots from stackerdb
+    pub fn get_signers(&self) -> Result<Vec<(StacksAddress, u64)>, ClientError> {
+        let contract_addr = boot_code_addr(self.chain_id == CHAIN_ID_MAINNET);
+        let contract_name = ContractName::from(SIGNERS_NAME);
+        let value = self.read_only_contract_call_with_retry(
+            &contract_addr,
+            &contract_name,
+            &ClarityName::from("stackerdb-get-signer-slots"),
+            &[],
+        )?;
+        let signer_slots = value.expect_result_ok()?.expect_list()?;
+        let mut signers = vec![];
+        for signer_slot in signer_slots {
+            let tuple = signer_slot.expect_tuple()?;
+            let principal = tuple.get("signer")?.clone().expect_principal()?;
+            let slots = u64::try_from(tuple.get("num-slots")?.clone().expect_u128()?)
+                .expect("BUG: Signers stacker db contract is corrupted");
+            let signer = if let PrincipalData::Standard(data) = principal {
+                (data.into(), slots)
+            } else {
+                panic!("BUG: Signers stacker db contract is corrupted");
```

Comment:
This code shouldn't panic. You might have gotten corrupted data over the network. Instead, you should just retry with backoff.

Reply:
Ah I see you want not just the call to the node to be on a retry but the deserialization as well. Will do :)